pkg/pillar/dpcmanager/dpc.go

```diff
@@ -222,12 +232,21 @@ func (m *DpcManager) ingestDPCList() (dpclPresentAtBoot bool) {
 	item, err := m.PubDevicePortConfigList.Get("global")
 	var storedDpcl types.DevicePortConfigList
 	if err != nil {
-		m.Log.Errorf("No global key for DevicePortConfigList")
-		dpclPresentAtBoot = false
+		m.Log.Errorf("Failed to load persisted DevicePortConfigList, trying backup: %v", err)
+		b, err := ioutil.ReadFile(backupDPCL)
+		if err == nil {
+			err = json.Unmarshal(b, &storedDpcl)
+		}
```

Comment:
I think it makes sense to replace with with calling a new pubsub function (TryBackup()?) which would do the stat of .bak, then if it exists copy from it to the normal name (so .bak stays should we get a power failure here), then redo the publications populate() function.

Reply:
Thinking some more, this can be completely hidden inside the pubsub's populate.

Reply:
Done, see updated PR description